docs/tildagonos-apps/guides/ui-widgets.md

```diff
@@ -0,0 +1,90 @@
+# UI Widget Overview
+
+You can use the following premade [`app_components`](guides/ui-widgets.md) to create user interfaces:
+
+## Menu
+
+The [`Menu`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/menu.py) component, allows you to create menus with which you can scroll through a list of options and select an item.
+
+## Notification
+
+The [`Notification`](https://github.com/emfcamp/badge-2024-software/blob/main/modules/app_components/notification.py) component allows you to create pop up notifications.
+
+### Example
+
+```python
+from app import App
+from app_components import Notification, clear_background
+from events.input import Buttons, BUTTON_TYPES
+
+
+class NotificationDemo(App):
+    def __init__(self):
+        self.notification = None
+        self.button_states = Buttons(self)
+
+    def update(self, delta):
+        if self.button_states.get(BUTTON_TYPES["CANCEL"]):
+            self.minimise()
+        if self.button_states.get(BUTTON_TYPES["DOWN"]):
+            self.notification = Notification("You pressed down!")
+
+    def draw(self, ctx):
+        clear_background(ctx)
+        if self.notification:
+            self.notification.draw(ctx)
+
+    def update(self, delta):
+        if self.notification:
+            self.notification.update(delta)
+
+```
+
+### Usage
+
+To use a notification:
+
+1. Import the `Notifiaction` component:
+
+    ```python
+    from app_components import Notification
+    ```
+
+2. Initialize the notification when in the `__init__` method of your app:
+
+    ```python
+    self.notification = None
+    ```
+
+3. Set the notification text in your app's code:
+
+    ```python
+    self.notification = Notification("This is a notification!")
+    ```
+
+4. Add the following lines in your `draw()` method to draw the notification when `self.notification` contains a notification:
+
+    ```python
+    # in def draw(serlf, ctx):
+    if self.notification:
+        self.notification.draw(ctx)
+    ```
+
+5. Add the following lines in your `update()` method to ??.
```

Comment:
unsure what the update method does

Reply:
Update is for letting the badge do work that isn't going to modify the UI. Here's where you'd handle responses to http requests for example - and you'd modify state potentially, or check "what's the signal strength of that ibeacon and do I need to set a flag so that the draw method does something next time it's called" - and that way the check isn't happening in the draw method and slowing it down.